docs/source/guide/zne-5-theory.myst

```diff
@@ -0,0 +1,118 @@
+devices with pulse-level access {cite}`Temme_2017_PRL, Kandala_2019_Nature`.
+The noise of the device can be altered by increasing the time over which pulses
+are implemented, as shown in the following diagram:
+
+```{figure} ../img/zne_pulse_stretching.png
+---
+width: 500
+name: figzne_pulse
+---
+The diagram demonstrates how pulse stretching is used to increase noise in a physical device for ZNE.
+```
+
+## Step 2: Extrapolate to the noiseless limit
+
+In both gate-model and pulse-model scenarios, let $\tau$ be a parameter quantifying
+noise level in the circuit and let $\tau^\prime = \lambda \tau$ the scaled noise level.
+
+For $\lambda = 1$, the input circuit remains unchanged
+since the noise level $\tau^\prime = \tau$ is the same as the noise of the physical device without
+any scaling.
+
+Let $\rho(\tau')$ be the state prepared by a noise scaled quantum circuit. The expectation
+value of an observable $A$ can be described as a function of the noise scaling parameter as follows:
+
+$$
+\langle E(\lambda) \rangle = \text{Tr}[\rho(\tau') A] = \text{Tr}[\rho(\lambda \tau) A]
+$$
+
+The idea of ZNE is that one can estimate the ideal expectation value $\langle E(\lambda=0) \rangle$, by measuring a range of different
+expectation values $\langle E(\lambda) \rangle$ for different values of $\lambda \ge 1$ and extrapolating
+to the zero-noise limit.
+
+In practice the extrapolation can be done as follows:
+  1) Assume $E(\lambda)\simeq f(\lambda; p_1, p_2, ... p_m)$, where $f$ is an *extrapolation model*, i.e., a function
+  of $\lambda$ depending on a set of real parameters $p_1, p_2, \dots, p_m$.
+  2) Fit the function $f$ to the measured noise-scaled expectation values, obtaining an optimal set of 
+  parameters $\tilde p_1, \tilde p_2, \dots \tilde p_m$.
+  3) Evaluate the corresponding zero-noise limit, i.e., $f(0; \tilde p_1, \tilde p_2, \dots \tilde p_m)$.
+
+Different choices of $f$, produce different extrapolations. Typical choices for $f$ are: a linear function, a polynomial, an exponential.
+For example, Richardson extrapolation, corresponds to the following polynomial model:
+
+
+$$
+f(\lambda; p_1, p_2, ... p_m) = p_1 + p_2 \lambda + p_3 \lambda^2 + \dots p_m \lambda^{m-1},
+$$
+
+where $m$ is equal to the number of data points in the fit (i.e. the number of noise scaled expectation values).
+ 
+where $S$ is the set of
+chosen scale factors, $n_{\lambda}$ is a coefficient dependent on the scale factor
+when the noisy expectation value is calculated after scaling the noise in the circuit.
+$\Delta$ is the variance error caused by choosing some extrapolation model.
+
+
+
+Different extrapolation methods can be used for $\langle E\rangle_{\rm ideal}$. A
+detailed discussion about the extrapolation methods is [available
+here](https://mitiq--1021.org.readthedocs.build/en/1021/guide/zne-3-options.html#extrapolation-methods-factory-objects).
```

Comment:
```suggestion
More details on how to apply different extrapolation methods with Mitiq can be found in [What additional options
are available in ZNE?](zne-3-options.myst).
```
This suggestion removes a previous sentence that is now outdated after the last changes and adds a link to the option
notebook using the name of the file instead of a URL.  The name of the file is more stable, the static URL may change and/or disappear.

Reply:
I think instead of linking the entire file, it's better to link the subsection. I was going to keep the PR url and edit it later once the PR was merged. But like you said the URL might disappear. 

So, another way to link that particular section will be to create a reference in `zne-3-options.myst`, unless there's some other way to do it.

Reply:
Adding the name of the section after the file name seems to work:  `(file_name.myst#name_of_the_section_with_lower_case_letters)`.  For example:
```
[What additional options are available in ZNE?](zne-3-options.myst#extrapolation-methods-factory-objects)
```
Creating an explicit reference is also fine but I didn't try it.